src/codemagic/apple/app_store_connect/provisioning/beta_groups.py

```diff
@@ -0,0 +1,65 @@
+from dataclasses import dataclass
+from typing import Optional
+from typing import Type
+from typing import Union
+
+from codemagic.apple.app_store_connect.resource_manager import ResourceManager
+from codemagic.apple.resources import BetaGroup
+from codemagic.apple.resources import Build
+from codemagic.apple.resources import ResourceId
+from codemagic.apple.resources import ResourceType
+
+
+class BetaGroups(ResourceManager[BetaGroup]):
+    """
+    https://developer.apple.com/documentation/appstoreconnectapi/betagroup
+    """
+
+    @property
+    def resource_type(self) -> Type[BetaGroup]:
+        return BetaGroup
+
+    @dataclass
+    class Filter(ResourceManager.Filter):
+        id: Optional[ResourceId] = None
+        name: Optional[str] = None
+        app: Optional[str] = None
+
+    def list(self, resource_filter: Filter = Filter()):
```

Comment:
Missing return type hint.
```suggestion
    def list(self, resource_filter: Filter = Filter()) -> List[BetaGroup]:
```